### ISSUE TITLE
Missing theme files that are not required

### ISSUE THREAD

**Issue by justintadlock** [WPTRT/theme-sniffer#134]
Theme Sniffer is erroneously requiring several theme files on https://github.com/WPTRT/theme-sniffer/blob/master/src/callback/class-run-sniffer-callback.php#L783

**comments.php**

This file is not required.  What's required is a call to the `comments_template()` function, which can reference any file name.

**functions.php**

Themes do not need a `functions.php`.

**screenshot.png**

I believe we also allow a `screenshot.jpg` at .ORG.  Other image file types may be allowed.  Need to check.

**Comment by justintadlock** [comment]
There's some crossover with this issue and https://github.com/WPTRT/theme-sniffer/issues/128.

**Comment by pattonwebz** [comment]
I have some work related to this issue inside of #129.

**Comment by timelsass** [comment]
@justintadlock do you know who can update the handbook reference - I thought it was weird `comments.php` was listed as a requirement and had updated a few themes to account for that.

**Comment by pattonwebz** [comment]
@timelsass I can update the page if we come up with a more correct wording for it. The current text is `Include comments.php (via comments_template()).` which I agree is incorrect.

**Comment by justintadlock** [comment]
I don't have access to the themes handbook (just the Make blog).  

@pattonwebz - Because the [handbook page](https://developer.wordpress.org/themes/release/required-theme-files/) is referring to required **files**, maybe it can simply be removed.  The requirement of `comments_template()` should be listed elsewhere.

Or, possibly have a sub-section under that of "maybe required" files and cover all the `/wp-includes/theme-compat` things that a theme may need to have if using certain templates/features.

**Comment by timelsass** [comment]
Yeah thanks for linking that @justintadlock  that's the page I was thinking of

**Comment by carolinan** [comment]
Doesn't the uploader also require sidebar.php? All 3 would need to require the same files, uploader, theme check, theme sniffer  (And not show errors for child themes).

**Comment by justintadlock** [comment]
> Doesn't the uploader also require sidebar.php?

No.  I have many themes without that file.

What you're probably thinking of is:  the sidebar check from TC is to see if `dynamic_sidebar()` is called in the scenario that a `register_sidebar()` call is found.
